### DISCUSSION TITLE
Difference between PacketRusher/lib/gtp5g and free5gc/gtp5g

### DISCUSSION THREAD

**Discussion by abousselmi** [HewlettPackard/PacketRusher#87] [General]
Hi @linouxis9 

Is there any difference between PacketRusher/lib/gtp5g and free5gc/gtp5g ? Why don't we use the official gtp5g project ?

Thanks !

**Comment by linouxis9** [comment]
Hi @abousselmi,

It was because I had initially planned to modify it, but never ended up doing it since we started switching to eBPF.
We kept it as it made the instructions simpler.
But if you are using it on Docker where the gtp5g is already installed for the free5gc's UPF, you may reuse it.

Thanks and cheers !
Valentin

**Reply by abousselmi** [reply]
Hi @linouxis9 

Thanks ! That's very clear 👍 

Cheers
